torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -1057,6 +1057,13 @@ struct IValuePacker {
   // That's what the TypePtr is for: it contains the information to do the
   // parsing. See torch::jit::toIValue for more information.
   static at::TypePtr packed_type() {
+#ifdef _WIN32
+    // NB: the if-constexpr usage triggers compilation errors on Windows
+    // with certain compiler settings
+    // (see https://github.com/pytorch/pytorch/pull/144707 for examples).
+    // It's not clear what the problem is, so we're going to ignore it for now.
+    TORCH_INTERNAL_ASSERT(false, "torch.compile not supported on Windows");
```

Comment:
I though torch.compile was supported on Windows, and somehow this only affects Windows builds, so may be a more selective ifdef would be good there

Reply:
I don't think torch.compile works on windows: https://github.com/pytorch/pytorch/issues/122094, but I've never tried it out. The main problem we had was that triton doesn't work on windows